### ISSUE TITLE
Issue with aws_cognito_user_pool_client URLs when they are not sorted alphabetically

### ISSUE THREAD

**Issue by arocki7** [terraform-providers/terraform-provider-aws#4550]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v0.11.7
+ provider.aws v1.14.1

### Affected Resource(s)

* aws_cognito_user_pool_client

### Terraform Configuration Files

```hcl
resource "aws_cognito_user_pool_client" "google-test" {
  name                                  = "google-test"
  user_pool_id                          = "${aws_cognito_user_pool.google-test.id}"
  generate_secret                       = false
  allowed_oauth_flows_user_pool_client  = true
  allowed_oauth_flows                   = ["code", "implicit"]
  allowed_oauth_scopes                  = ["openid", "phone", "aws.cognito.signin.user.admin", "profile", "email"]
  callback_urls                         = ["http://localhost:8080", "https://www.google.com", "https://business.google.com"]
  explicit_auth_flows                   = ["ADMIN_NO_SRP_AUTH"]
  logout_urls                           = ["http://localhost:8080", "https://www.google.com", "https://business.google.com"]
  read_attributes                       = ["given_name","email_verified","zoneinfo","website","preferred_username","name","locale","phone_number","family_name","birthdate","middle_name","phone_number_verified","custom:provider","profile","picture","address","gender","updated_at","nickname","email"]
  write_attributes                      = ["given_name","zoneinfo","website","preferred_username","name","locale","phone_number","family_name","birthdate","middle_name","custom:provider","profile","picture","address","gender","updated_at","nickname","email"]
  supported_identity_providers          = ["COGNITO","Facebook","Google"]
}
```

### Debug Output

Output in AWS Cognito User pool client URLs:

```
http://localhost:8080, https://business.google.com, https://www.google.com
```

### Panic Output

```
Nil
```

### Expected Behavior

```
No changes. Infrastructure is up-to-date.

This means that Terraform did not detect any differences between your
configuration and real physical resources that exist. As a result, no
actions need to be performed.
```

### Actual Behavior

The URLs in AWS Cognito user pool client are sorted alphabetically. So, it conflicts with terraform coding which are not sorted alphabetically.

```
An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  ~ aws_cognito_user_pool_client.sagepass-prod
      callback_urls.1: "https://business.google.com" => "https://www.google.com"
      callback_urls.2: "https://www.google.com" => "https://business.google.com"
      logout_urls.1:   "https://business.google.com" => "https://www.google.com"
      logout_urls.2:   "https://www.google.com" => "https://business.google.com"


Plan: 0 to add, 1 to change, 0 to destroy.
```

### Steps to Reproduce

1. Write the callback_urls & logout_urls in way where it is not sorted alphabetically.
2. `terraform plan`

### Important Factoids

Nil

### References

No References

**Comment by ericwong3** [comment]
I just encountered this problem and a temporarily solution is to use `sort()`, .e.g.
```
resource "aws_cognito_user_pool_client" "admin_pool_client" {
  ...
  callback_urls = sort(var.admin_callback_urls)
  logout_urls = sort(var.admin_logout_urls)
}
```
This prevents TF from treating it as a change constantly

**Comment by DrFaust92** [comment]
I was able to recreate this by changing the order of callback_urls in acceptance tests and submitted a PR to address this

**Comment by bflad** [comment]
The fix for this has been merged and will release with version 2.54.0 of the Terraform AWS Provider, later this week. Thanks to @DrFaust92 for the implementation. 👍
